### PR TITLE
Multi language labels

### PR DESCRIPTION
**Pull request by gucio321**
Hi there,
Labels in Main Menu should now depends on string tables.
I don't say, that's a perfect solution of #944, but it works for now.
Let me know, is main_menu.go right place to put this "table interpretation", or maybe better to put it into d2resource?

### CODE REVIEW

File: d2game/d2gamescreen/main_menu.go [hunk 1]
```diff
@@ -83,6 +83,142 @@ type BuildInfo struct {
...
+		1704, // nec description
+		1709, // barb description
+		1710, // sorc description
+		1711, // pal description
+
+		/* these items are for battle.net multiplayer
+		labels and are not used yet
+		in addition, as many elements as the value
+		of the highest modifier must be replaced*/
+		1712, // not used here
```
Comment (gravestench) [inline d2game/d2gamescreen/main_menu.go:199]: Why put this here, then? is this value used?

File: d2common/d2resource/languages_map.go [hunk 2]
```diff
@@ -49,3 +49,30 @@ func GetFontCharset(language string) string {
...
+		"RUS": 0, // (Russian) // checked
+	}
+}
+
+// GetLabelModifier returns modificator for language
+func GetLabelModifier(language string) int {
+	mod := getModifiers()
+
+	return mod[language]
+}
```
Comment (gravestench) [inline d2common/d2resource/languages_map.go:78]: replace all `modificator` with `modifier`

~~How are these integers used... ?~~ I think you should state that this table is used as an index offset for string keys of the form `#1234`

also, you could use just one function here:
```golang
// GetLabelModifier returns modificator for language
func GetLabelModifier(language string) int {
	modifiers := map[string]int{
		"ENG": 0, // (English) // checked
		"ESP": 0, // (Spanish)
		"DEU": 0, // (German) // checked
		"FRA": 0, // (French)
		"POR": 0, // (Portuguese)
		"ITA": 0, // (Italian)
		"JPN": 0, // (Japanese)
		"KOR": 0, // (Korean)
		"SIN": 0, //
		"CHI": 0, // (Chinese)
		"POL": 1, // (Polish) // checked
		"RUS": 0, // (Russian) // checked
	}

	return modifiers[language]
}
```

File: d2game/d2gamescreen/main_menu.go (hunk 1 above)
Comment (gucio321) [reply d2game/d2gamescreen/main_menu.go:199]: sorry, wrong comment her

File: d2game/d2gamescreen/main_menu.go [hunk 3]
```diff
@@ -234,6 +374,10 @@ func (v *MainMenu) loadBackgroundSprites() {
 	v.serverIPBackground.SetPosition(serverIPbackgroundX, serverIPbackgroundY)
 }
 
+func translateLabel(label int, lng string, asset *d2asset.AssetManager) string {
+	return asset.TranslateString(fmt.Sprintf("#%d", baseLabelNumbers(label+d2resource.GetLabelModifier(lng))))
+}
+
```
Comment (gravestench) [inline d2game/d2gamescreen/main_menu.go:380]: I don't think that this logic belongs in the main menu. Retrieving the correct string for the current locale should be something that the asset manager deals with, not the main menu.

File: d2game/d2gamescreen/main_menu.go [hunk 4]
```diff
@@ -92,6 +162,7 @@ func CreateMainMenu(
 	audioProvider d2interface.AudioProvider,
 	ui *d2ui.UIManager,
 	buildInfo BuildInfo,
+	lng string,
```
Comment (gravestench) [inline d2game/d2gamescreen/main_menu.go:165]: is this used anymore?

Comment (gucio321) [reply d2game/d2gamescreen/main_menu.go:165]: no, I forgot to remove.

File: d2common/d2util/stringutils.go [hunk 5]
```diff
@@ -135,11 +135,6 @@ func SplitIntoLinesWithMaxWidth(fullSentence string, maxChars int) []string {
 	return lines
 }
 
-// SplitIntoLinesWithMaxWidthOneLine does the same as SplitIntoLinesWithMaxWidth but return string with newline char
-func SplitIntoLinesWithMaxWidthOneLine(s string, l int) string {
-	return strings.Join(SplitIntoLinesWithMaxWidth(s, l), "\n")
-}
-
```
Comment (gravestench) [inline d2common/d2util/stringutils.go:142]: Thank you fr removing this :)

File: d2game/d2gamescreen/main_menu.go [hunk 6]
```diff
@@ -83,6 +84,76 @@ type BuildInfo struct {
...
+	hardCoreLabel
+	selectHeroClassLabel
+	amazonDescr
+	necromancerDescr
+	barbarianDescr
+	sorceressDescr
+	paladinDescr
+
+	//nolint:deadcode,varcheck,unused // will be used, or is need
+	notUsed1 // this position isn't used here.
```
Comment (gravestench) [inline d2game/d2gamescreen/main_menu.go:134]: i think you can put `_` if you want to skip an index here

File: d2game/d2gamescreen/main_menu.go [hunk 7]
```diff
@@ -83,6 +84,76 @@ type BuildInfo struct {
...
+	epilogueLabel
+	selectCinematicLabel
+
+	// multiplayer menu labels
+	//nolint:deadcode,varcheck,unused // will be used, or is need
+	openBattleNetLabeL
+	tcpIPGameLabel
+	tcpIPOptionsLabel
+	tcpIPHostGameLabel
+	tcpIPJoinGameLabel
```
Comment (gravestench) [inline d2game/d2gamescreen/main_menu.go:113]: This is ok for now, but i do think that this enum belongs elsewhere.

**Review by gravestench — APPROVED** [review]

**Pull request merged**